src/main/java/org/zalando/planb/revocation/persistence/CassandraAuthorizationRuleStore.java

```diff
@@ -0,0 +1,112 @@
+
+    private static final long RELOAD_IN_MS =  1000 * 60 * 10; // 10 minutes
+
+    private List<AuthorizationRule> inMemoryRuleStore = Collections.emptyList();
+
+    private final Session session;
+
+    private final PreparedStatement getRules;
+
+    private final PreparedStatement insertRule;
+
+    private static final String AUTHORIZATION_TABLE = "authorization_rule";
+    private static final String REQUIRED_USER_CLAIMS = "required_user_claims";
+    private static final String ALLOWED_REVOCATION_CLAIMS = "allowed_revocation_claims";
+    private static final String CREATED_BY = "created_by";
+    private static final String LAST_MODIFIED_BY = "last_modified_by";
+    private static final String UUID = "uuid";
+
+    private static final RegularStatement INSERT_AUTHORIZATION = QueryBuilder.insertInto(AUTHORIZATION_TABLE)
+            .value(UUID, now())
+            .value(REQUIRED_USER_CLAIMS, bindMarker())
+            .value(ALLOWED_REVOCATION_CLAIMS, bindMarker())
+            .value(CREATED_BY, bindMarker())
+            .value(LAST_MODIFIED_BY, bindMarker());
+    private static final RegularStatement SELECT_AUTHORIZATION = QueryBuilder.select()
+            .column(REQUIRED_USER_CLAIMS)
+            .column(ALLOWED_REVOCATION_CLAIMS)
+            .from(AUTHORIZATION_TABLE);
+    private static final RegularStatement CLEANUP_AUTHORIZATION = QueryBuilder.truncate(AUTHORIZATION_TABLE);
+
+    public CassandraAuthorizationRuleStore(final Session session, final ConsistencyLevel read, final ConsistencyLevel write) {
+        this.session = session;
+        getRules = session.prepare(SELECT_AUTHORIZATION).setConsistencyLevel(read);
+        insertRule = session.prepare(INSERT_AUTHORIZATION).setConsistencyLevel(write);
+    }
+
+    @PostConstruct
+    void initializeAuthorizationRuleStore() throws Exception {
+        loadAuthorizationRuleStore();
+        log.debug("Authorization rule store initialized");
+    }
+
+    @Scheduled(fixedDelay = RELOAD_IN_MS, initialDelay = RELOAD_IN_MS)
+    void loadAuthorizationRuleStore() {
+        inMemoryRuleStore = Optional.ofNullable(getRules.bind())
+                .map(session::execute)
+                .map(ResultSet::all)
+                .map(this::toAuthorizationRules)
+                .orElse(Collections.emptyList());
+    }
+
+    @Override
+    public Collection<AuthorizationRule> retrieveByMatchingAllowedClaims(AuthorizationRule authorizationRule) {
+        return findMatchingRulesByAllowedClaims(inMemoryRuleStore, authorizationRule);
+    }
+
+    @Override
+    public void store(AuthorizationRule authorizationRule) {
+        final BoundStatement insert = insertRule.bind()
+                                    .setMap(REQUIRED_USER_CLAIMS, authorizationRule.requiredUserClaims())
+                                    .setMap(ALLOWED_REVOCATION_CLAIMS, authorizationRule.allowedRevocationClaims())
+                                    .setString(CREATED_BY, null)
```

Comment:
There is no REST API yet to store authorizations, so it's manually inserted. This method is only used by the test-cases.